starlite/asgi/routing_trie/mapping.py

```diff
@@ -56,12 +52,40 @@ def add_mount_route(
     return current_node
 
 
+def validate_path_parameter(
+    current_node: "RouteTrieNode", param_def: PathParameterDefinition, route: "RouteType"
+) -> PathParameterDefinition:
+    """Validate that path parameter can be added to the current node.
+
+    Args:
+        current_node: parent node of path parameter component.
+        param_def: the path parameter definition.
+        route: route that is being parsed to the routing trie.
+
+    Raises:
+        ImproperlyConfiguredException: if a path parameter of same type exists as a child of `current_node` that is not
+            identical to `param_def`.
+    """
+    child_param_with_same_type = current_node.child_path_parameter_type_map.get(param_def.type)
+    if child_param_with_same_type:
+        if param_def != child_param_with_same_type:
+            existing_node = current_node.children[child_param_with_same_type]
+            raise ImproperlyConfiguredException(
+                f"Path parameter '{{{param_def.full}}}' for path '{route.path}' conflicts with parameter "
+                f"path '{existing_node.path}'."
+            )
```

Comment:
We could probably go a bit further with this validation, e.g., if there has been a broader type already registered, e.g., a `float` param registered before an `int` param, or a `str` param registered before a `uuid` param, we could raise an error.